util/list_keyboards.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+# Temporary shell script to find keyboards
+#
+# This allows us to exclude keyboards by including a .noci file.
+
+find keyboards -type f -name rules.mk | grep -v keymaps | while read keyboard; do
+	keyboard=$(echo $keyboard | sed 's!keyboards/\(.*\)/rules.mk!\1!') 
+
+	[ ! -e "keyboards/${keyboard}/.noci" ] && echo "$keyboard"
```

Comment:
As this is called at the start of any make command, if you now run `make clueboard/66/rev1:default` it stops you from building locally with the following output:
```console
QMK Firmware 0.11.3
make: *** No rule to make target 'clueboard/66/rev1:default'. Stop.
|
|  QMK's make format recently changed to use folder locations and colons:
|     make project_folder:keymap[:target]
|  Examples:
|     make dz60:default
|     make planck/rev6:default:flash
|
```

Off the top of my head, the check would have to be later maybe within PARSE_KEYBOARD if `ifeq ($$(KEYBOARD_RULE),all)`

Reply:
I've "fixed" this with 532ef0dc4 but the solution isn't very elegant. We're generating a list of keyboards twice which isn't great, but for some reason if I define `CI_KEYBOARDS` inside of `PARSE_ALL_KEYBOARDS` it ends up with an empty list, and I don't know why.

Reply:
ce9736a23: if you don't use a variable you don't have to worry about where it's defined.

![image](https://user-images.githubusercontent.com/9553982/101990981-7c018980-3c5e-11eb-8583-e6fd01856e77.png)